src/contract_client.rs

```diff
@@ -682,14 +683,15 @@ pub enum InvokeContractOutcome {
     Failure(RejectedTransaction),
```

Comment:
@limemloh wrote a suggestion somewhere to have a 
```rust
pub fn success() -> Result<SimulatedTransaction, RejectedTransaction>
```
method for this type, which would let users work with the type as errors as well. I think this is a pretty good compromise as it would facilitate more ways of easily handling the different outcomes.

What do you think?